crates/fuel-core/src/executor.rs

```diff
@@ -467,15 +472,21 @@ mod tests {
                 },
                 changes,
             ) = producer
-                .produce_without_commit_with_source(Components {
-                    header_to_produce: header,
-                    transactions_source: OnceTransactionsSource::new(vec![
-                        script.into(),
-                        invalid_duplicate_tx,
-                    ]),
-                    gas_price: price,
-                    coinbase_recipient: recipient,
-                })
+                .produce_without_commit_with_source(
+                    Components {
+                        header_to_produce: header,
+                        transactions_source: OnceTransactionsSource::new(vec![
+                            script.into(),
+                            invalid_duplicate_tx,
+                        ]),
+                        gas_price: price,
+                        coinbase_recipient: recipient,
+                    },
+                    TimeoutOnlyTxWaiter,
+                    TransparentPreconfirmationSender,
+                )
+                .now_or_never()
+                .unwrap()
```

Comment:
It is a lot of places where we need to use `now_or_never`. It would be better if we added a new helper sync function that we can reuse in these tests

Reply:
Done.